butlarr/services/ext.py

```diff
@@ -112,3 +114,78 @@ async def cmd_help(self, update, context, args):
             response_message += f"\n - `/{self.commands[0]} {cmd} {escape_markdownv2_chars(pattern)}` \t _{escape_markdownv2_chars(desc)}_"
 
         return await update.message.reply_text(response_message, parse_mode="Markdown")
+    
+    def media_caption(self, item, overview=True):
+        caption = f"{item['title']} "
+        if item["year"] and str(item["year"]) not in item["title"]:
+            caption += f"({item['year']}) "
+
+        if item["runtime"]:
+            caption += f"{item['runtime']}min "
+
+        caption += f"- {item['status'].title()}"
+        if overview:
+            caption += f"\n\n{item.get('overview', '')}"
+
+        caption = caption[0:1024]
+        return caption
+
+    def load_addons(self):
+        from ..config.services import SERVICES
+        logger.info(f"Loading {self.name} addons")
+        addons = []
+        for addon in self.addons:
+            for addon_service in SERVICES:
+                if addon_service.name == addon.get("service_name"):
+                    if addon_service.arr_variant in self.supported_addons:
+                        addons.append(addon_service)
+                        logger.info(f"Addon {addon_service.name} loaded")
+                    else:
+                        assert False, f"Unsupported addon service type {addon.get('type')}!"
+                        return False
+        self.addons = addons
+        logger.debug(f"{self.name} service loaded Addons: {str(self.addons)}")
+    
+
+@dataclass(frozen=True)
+class ParentState:
+    service: ArrService = None
+    state: any = None
+    menu: str = None
+
+class Addon:
+    service: ArrService = None
+    state: any = None 
+    menu: str = None
```

Comment:
It is confusing that this class does double duty by providing the static function wrap definitions and the new addon_buttons interface. Move the functions out of there.
Maybe I should think about splitting the ext folder up into different files for clarity too :)

Reply:
The addon class is supposed to provide all the "extras" a service need to be also used as and addon (so, the buttons, the wrappers and the parent state) I don't really think that it really is a double duty.

The parent state thing for sure can be changed in some way, for now i replaced the 3 vars with a "parent" object:

```python
class ParentState:
    service: ArrService = None
    state: any = None
    menu: str = None

class Addon:
    parent: ParentState = None
    supported_services = []
```